### DISCUSSION TITLE
i2c LCD not working while playing audio

### DISCUSSION THREAD

**Discussion by Udaymakxenia** [pschatzmann/arduino-audio-tools#840] [Q&A]
Hi Phil I was trying out a basic sketch where the lcd prints data while the audio plays . 
But if the decoder starts the lcd will not print anything after that and gets stuck. 
Could you tell me if this is something related to the two devices using the same i2c bus .
I have a ESP32A1s (AC101) .
```/**
   @file streams-url_mp3-audiokit.ino
   @author Phil Schatzmann
   @brief decode MP3 stream from url and output it on I2S on audiokit
   @version 0.1
   @date 2021-96-25

   @copyright Copyright (c) 2021
*/

// install https://github.com/pschatzmann/arduino-libhelix.git
#include <Wire.h>
#include <LiquidCrystal_I2C.h>
LiquidCrystal_I2C lcd(0x27, 16, 2);

#include "AudioTools.h"
#include "AudioCodecs/CodecMP3Helix.h"
#include "AudioLibs/AudioKit.h"
int i = 0;

URLStream url("*****", "******"); // or replace with ICYStream to get metadata
AudioKitStream i2s; // final output of decoded stream
EncodedAudioStream dec(&i2s, new MP3DecoderHelix()); // Decoding stream
StreamCopy copier(dec, url); // copy url to decoder


void setup() {
  Serial.begin(115200);
  lcd.begin(21,22);
  lcd.backlight();
  lcd.setCursor(0, 1);
  lcd.print("SOC WAP v1.1");
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // setup i2s
  auto config = i2s.defaultConfig(TX_MODE);
  i2s.begin(config);

  // setup I2S based on sampling rate provided by decoder
  dec.begin();

  // mp3 radio
  url.begin("http://stream.srg-ssr.ch/m/rsj/mp3_128", "audio/mp3");

}

void loop() {
  lcd.print("Playing");
  lcd.print(i);
  copier.copy();
  i++;
}
```
I tried individual codes and it works but when I merge them the audio plays well but the lcd hangs.
I tried disabling USE AUDIOKIT_WIRE in the library, did not work.
Could you give some suggestions.

**Comment by pschatzmann** [comment]
I have the impression that you use Wire for the LCD and Wire1 for the AudioKit on the same pins: this can't work!
The solution should be one of the following options:
- use different pins
- change the audiokit config to use Wire instead of Wire1
- change the lcd to use Wire1 instead of Wire

**Reply by Udaymakxenia** [reply]
I have used pin 21 and 22 for the lcd
I tried changing audiokit config to use Wire1 instead of Wire
I also tried using Wire 1 for lcd as well.
Still the same result.

**Reply by pschatzmann** [reply]
Have you tried to use the same the LCD and the  AC101 on the same Bus with the pins defined in the AudioKit Wiki?
I never tried to connect a LCD so I am not sure what the issue could be...

**Reply by Udaymakxenia** [reply]
![image](https://github.com/pschatzmann/arduino-audio-tools/assets/109062412/cf3451bf-b928-4d5f-ab5d-a6b961ed5f97)
Are you asking if i tried connecting lcd to 33 and 32? 
6 corresponds to ac101

**Comment by pschatzmann** [comment]
Yes, but you also need to make sure that both are using either the same port: either Wire or Wire1

**Reply by Udaymakxenia** [reply]
As these are internal i would have to remove the shield I suppose?

**Reply by pschatzmann** [reply]
No you can change the audiokit config to use Wire or you can indicate in the lcd.begin to use Wire1

**Reply by Udaymakxenia** [reply]
I was also getting an i2c driver install error when i play the audio.
But the audio plays fine.

**Reply by Udaymakxenia** [reply]
> No you can change the audiokit config to use Wire or you can indicate in the lcd.begin to use Wire1

okk will try this

**Reply by pschatzmann** [reply]
Oh, my mistake - I just see that the 32 and 33 are not made available to connect from outside.
So a shared I2C is not really an option: So you need to make sure to use different ports

**Reply by Udaymakxenia** [reply]
> Oh, my mistake - I just see that the 32 and 33 are not made available to connect from outside. So a shared I2S is not really an option: So you need to make sure to use different ports

okk.

**Reply by Udaymakxenia** [reply]
> I was also getting an i2c driver install error when i play the audio. But the audio plays fine.

any comments on this?

**Comment by pschatzmann** [comment]
I think the issues are your selected pins: they are in use
#define PA_ENABLE_GPIO GPIO_NUM_21
#define GREEN_LED_GPIO GPIO_NUM_22

see https://github.com/pschatzmann/arduino-audiokit/blob/main/src/audio_board/ai_thinker_ac101/board_def.h

**Reply by Udaymakxenia** [reply]
But I am not using either PA ENABLE or GREEN LED.

**Reply by Udaymakxenia** [reply]
Should I disable this somewhere in the library?

**Reply by pschatzmann** [reply]
Yes, but the I2S Audiokit output is!

**Reply by pschatzmann** [reply]
No you need to select some pins which are not used!

**Reply by Udaymakxenia** [reply]
So it means I can't disable the PA enable right.

**Reply by pschatzmann** [reply]
If the PA is disabled (the Power Amplifier is low) you don't get any audio.
If you have some random signal on this pin, this is asking for trouble

**Reply by Udaymakxenia** [reply]
ok got it thanks . I will try this and let you know. Thanks for the help.

**Reply by Udaymakxenia** [reply]
Hey Phil tried changing pins but still the same. The lcd does not print anything ones the audio starts.
